### PR TITLE
CDF-16344 Infer project from the API key

### PR DESCRIPTION
**Pull request by emelvarol-cognite**

### CODE REVIEW

**Comment by emelvarol-cognite** [comment]
I'll add a small test for the new function.

File: tests/test_run.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def test_watch_only(
     cli_result_list = from_table(cli_result.output)
     assert "Failed" in cli_result_list[2]
     assert "SELECT 'asd' AS externalId 'asd' AS name" == " ".join(cli_result_list[4])
-    assert "mismatched input ''asd'' expecting {<EOF>, ';'}(line 1, pos 27)" == " ".join(cli_result_list[6])
+    assert "Syntax error at or near ''asd''(line 1, pos 27)" == " ".join(cli_result_list[6])
```
Comment (silvavelosa) [inline tests/test_run.py:74]: is this an error msg from the api? relaying on server side strings can make these tests flaky... any chance of testing for an http code, a status, of something with lower chance of changing?

**Review by silvavelosa — APPROVED** [review]
Nice! Just a small comment (no big deal... feel free to ignore it if a client is being blocked by this)

File: tests/test_run.py (hunk 1 above)
Comment (emelvarol-cognite) [reply tests/test_run.py:74]: It is not a big deal, the error returned from failing transformation changed for some reason. I check if we propagate the error nicely, so I think worst case we change it again in cli prs.

Comment (emelvarol-cognite) [reply tests/test_run.py:74]: this is not sdk, so it doesn't block anyone else.

**Pull request merged**